pkg/client/restclient/url_utils.go

```diff
@@ -27,27 +27,46 @@ import (
 // DefaultServerURL converts a host, host:port, or URL string to the default base server API path
 // to use with a Client at a given API version following the standard conventions for a
 // Kubernetes API.
-func DefaultServerURL(host, apiPath string, groupVersion unversioned.GroupVersion, defaultTLS bool) (*url.URL, string, error) {
+func DefaultServerURL(host string, defaultTLS bool) (*url.URL, error) {
 	if host == "" {
-		return nil, "", fmt.Errorf("host must be a URL or a host:port pair")
+		return nil, fmt.Errorf("host must be a URL or a host:port pair")
 	}
 	base := host
 	hostURL, err := url.Parse(base)
 	if err != nil {
-		return nil, "", err
+		return nil, err
 	}
-	if hostURL.Scheme == "" || hostURL.Host == "" {
+	if hostURL.Scheme == "" {
 		scheme := "http://"
 		if defaultTLS {
 			scheme = "https://"
 		}
 		hostURL, err = url.Parse(scheme + base)
 		if err != nil {
-			return nil, "", err
+			return nil, err
 		}
 		if hostURL.Path != "" && hostURL.Path != "/" {
-			return nil, "", fmt.Errorf("host must be a URL or a host:port pair: %q", base)
+			return nil, fmt.Errorf("host must be a URL or a host:port pair: %q", base)
 		}
+
+	}
+	return hostURL, nil
+
+}
+
+func DefaultServerURLsForHosts(hosts []string, apiPath string, groupVersion unversioned.GroupVersion, defaultTLS bool) ([]*url.URL, string, error) {
```

Comment:
Needs a comment.  In particular to support moving the localhost default into this function.